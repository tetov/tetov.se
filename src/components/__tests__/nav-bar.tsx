import React from "react";
import renderer from "react-test-renderer";
import NavBar from "src/components/nav-bar";

describe("NavBar", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NavBar pathname="/" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
