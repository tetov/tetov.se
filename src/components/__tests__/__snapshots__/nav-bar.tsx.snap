// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<nav
  className="text-center md:justify-between mb-8 border-b-2 border-dark-purple"
  data-headlessui-state=""
>
  <div
    className="mx-auto px-6 sm:px-6 lg:px-8"
  >
    <div
      className="relative flex h-16 items-center justify-start gap-x-6"
    >
      <div
        className="flex flex-shrink-0"
      >
        <a
          aria-hidden={true}
          href="/"
        >
          <test-file-stub
            className="h-8 w-auto flex flex-shrink-0"
          />
        </a>
      </div>
      <div
        className="flex items-start "
      >
        <a
          href="/"
        >
          <span
            className="text-xl font-bold tracking-tighter leading-tight whitespace-nowrap"
          >
            IÃ°unn Gunda
          </span>
        </a>
      </div>
      <button
        aria-expanded={false}
        autoFocus={false}
        className="sm:hidden ml-auto rounded-md p-2 text-gray-700 dark:text-gray-400 "
        data-headlessui-state=""
        id="headlessui-disclosure-button-:r0:"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPointerDown={[Function]}
        onPointerUp={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Open navigation menu
        </span>
        <div
          aria-hidden={true}
          className="block w-5 left-1/2 top-1/2 transform -translate-x-1/2 -translate-y-1/2"
        >
          <span
            className="block absolute h-0.5 w-5 bg-current transform transition duration-500 ease-in-out ui-open:rotate-45 ui-not-open:-translate-y-1.5"
          />
          <span
            className="block absolute h-0.5 w-5 bg-current transform transition duration-500 ease-in-out ui-open:opacity-0"
          />
          <span
            className="block absolute h-0.5 w-5 bg-current transform transition duration-500 ease-in-out ui-open:-rotate-45 ui-not-open:translate-y-1.5"
          />
        </div>
      </button>
      <div
        className="hidden sm:block ml-auto"
      >
        <div
          className="flex space-x-4"
        >
          <ul
            className="flex font-medium"
          >
            <li
              className="flex px-3 py-2"
            >
              <a
                aria-current="page"
                className="bg-dark-purple text-dark-text px-3 py-2"
                href="/"
              >
                Projects
              </a>
            </li>
            <li
              className="flex px-3 py-2"
            >
              <a
                className="hover:text-purple px-3 py-2"
                href="/posts"
              >
                Posts
              </a>
            </li>
            <li
              className="flex px-3 py-2"
            >
              <a
                className="hover:text-purple px-3 py-2"
                href="/about"
              >
                About
              </a>
            </li>
            <li
              className="flex px-3 py-2"
            >
              <a
                className="hover:text-purple px-3 py-2"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`NavLinks renders correctly horizontal 1`] = `
<ul
  className="flex font-medium"
>
  <li
    className="flex px-3 py-2"
  >
    <a
      className="hover:text-purple px-3 py-2"
      href="https://test1.test"
    >
      test1
    </a>
  </li>
  <li
    className="flex px-3 py-2"
  >
    <a
      className="hover:text-purple px-3 py-2"
      href="https://test2.test"
    >
      test2
    </a>
  </li>
</ul>
`;

exports[`NavLinks renders correctly vertical 1`] = `
<ul
  className="font-medium"
>
  <li
    className="px-3 py-2"
  >
    <a
      className="hover:text-purple px-3 py-2"
      href="https://test1.test"
    >
      test1
    </a>
  </li>
  <li
    className="px-3 py-2"
  >
    <a
      className="hover:text-purple px-3 py-2"
      href="https://test2.test"
    >
      test2
    </a>
  </li>
</ul>
`;
